plugins/funind/functional_principles_proofs.mli

```diff
@@ -1,19 +1,26 @@
 open Names
 
 val prove_princ_for_struct :
-  Evd.evar_map ref ->
-  bool ->
-  int -> Constant.t array -> EConstr.constr array -> int -> Tacmach.tactic
-
+     Evd.evar_map ref
+  -> bool
+  -> int
+  -> Constant.t array
+  -> EConstr.constr array
+  -> int
+  -> Tacmach.tactic
 
 val prove_principle_for_gen :
-  Constant.t * Constant.t * Constant.t -> (* name of the function, the functional and the fixpoint equation *)
-  Indfun_common.tcc_lemma_value ref -> (* a pointer to the obligation proofs lemma *)
-  bool -> (* is that function uses measure *)
-  int -> (* the number of recursive argument *)
-  EConstr.types -> (* the type of the recursive argument *)
-  EConstr.constr -> (* the wf relation used to prove the function *)
-  Tacmach.tactic
-
-
+     Constant.t * Constant.t * Constant.t
+  -> (* name of the function, the functional and the fixpoint equation *)
```

Comment:
I'm not sure about the mli changes, some manual attention is needed because it makes the comments look misplaced at least.
Having the arrow at the start of the line is not bad but it doesn't match the rest of the code, is there an option to change it?

Reply:
Not in the released version, you can use ` [@@@ocamlformat "disable=false"]` in that part.

I have switched to that style on my projects tho, it makes named arguments and `odoc` pretty convenient.

Reply:
Note that ocamldoc style is as done by `ocamlformat`, comment below the argument.

Reply:
I'd prefer the comment before the arrow though. I think it would produce something like
~~~ocaml
val bla : 
  foo
  (** it's a foo *)
  -> bar
  (** bla makes a bar *)
~~~

Reply:
Isn't this case a problem of the original code indeed? I guess ocmalformat cannot move comments thru arrow-boundaries.